agents/agents-features/agents-features-snapshot/README.md

```diff
@@ -0,0 +1,194 @@
+
+### Storage Providers
+
+AgentCheckpoint uses storage providers to save and retrieve checkpoints. The framework includes several built-in providers:
+
+- `InMemoryAgentCheckpointStorageProvider`: Stores checkpoints in memory (lost when the application restarts)
+- `FileAgentCheckpointStorageProvider`: Persists checkpoints to the file system
+- `NoAgentCheckpointStorageProvider`: A no-op implementation (default) that doesn't store checkpoints
+
+You can also implement custom storage providers by implementing the `AgentCheckpointStorageProvider` interface.
+
+### Continuous Persistence
+
+The feature can be configured to automatically create checkpoints after each node execution, ensuring that the agent's state is continuously persisted and can be recovered at any point.
+
+## Installation
+
+To use the AgentCheckpoint feature, you need to add it to your agent's configuration:
+
+```kotlin
+val agent = aiAgent {
+    // ... other configuration
+    
+    features {
+        install(AgentCheckpoint) {
+            // Configure the feature
+            snapshotProvider(InMemoryAgentCheckpointStorageProvider())
+            continuouslyPersistent() // Optional: create checkpoints after each node
+        }
+    }
+}
+```
+
+## Configuration Options
+
+The AgentCheckpoint feature has two main configuration options:
+
+### Storage Provider
+
+Set the storage provider that will be used to save and retrieve checkpoints:
+
+```kotlin
+install(AgentCheckpoint) {
+    snapshotProvider(InMemoryAgentCheckpointStorageProvider())
+}
+```
+
+Available built-in providers:
+- `InMemoryAgentCheckpointStorageProvider`: In-memory storage (non-persistent)
+- `FileAgentCheckpointStorageProvider`: File-based storage
+- `NoAgentCheckpointStorageProvider`: No-op provider (default)
+
+### Continuous Persistence
+
+Enable automatic checkpoint creation after each node execution:
+
+```kotlin
+install(AgentCheckpoint) {
+    continuouslyPersistent()
+}
+```
+
+When enabled, the agent will automatically create a checkpoint after each node is executed, allowing for fine-grained recovery.
+
+## Basic Usage
+
+### Creating a Checkpoint
+
+Create a checkpoint at a specific point in your agent's execution:
+
+```kotlin
+suspend fun example(context: AIAgentContextBase) {
+    // Create a checkpoint with the current state
+    val checkpoint = context.checkpoint().createCheckpoint(
+        agentId = context.id,
+        agentContext = context,
+        nodeId = "current-node-id",
+        lastInput = inputData
+    )
+    
+    // The checkpoint ID can be stored for later use
+    val checkpointId = checkpoint.checkpointId
+}
+```
+
+### Restoring from a Checkpoint
+
+Restore an agent's state from a specific checkpoint:
+
+```kotlin
+suspend fun example(context: AIAgentContextBase, checkpointId: String) {
+    // Roll back to a specific checkpoint
+    context.checkpoint().rollbackToCheckpoint(checkpointId, context)
```

Comment:
If we change the feature name to `checkpoints`, it will be also `checkpoints()` :)